core/src/Kokkos_Core.hpp

```diff
@@ -140,6 +140,9 @@ void fence(const std::string& name /*= "Kokkos::fence: Unnamed Global Fence"*/);
 /** \brief Print "Bill of Materials" */
 void print_configuration(std::ostream& os, bool verbose = false);
 
+// Free and total memory on the device
+std::pair<std::size_t, std::size_t> device_memory_info();
```

Comment:
What are your objections to
```suggestion
// Free and total memory on the device
std::pair<std::size_t, std::size_t> device_memory_info(const Kokkos::DefaultExecutionSpace& exec);
```
possibly defaulted to support multi-GPU setups?